pkg/checker/checker.go

```diff
@@ -42,3 +43,33 @@ func Build(cfg *config.CheckerConfig, kubeClient kubernetes.Interface) (Checker,
 	}
 	return builder(cfg, kubeClient)
 }
+
+// DefaultRecordResult increments the result counter for a specific checker run.
+// If err is not nil, it records a run error (unknown status).
+// If result is not nil, it records the status from the result.
+func DefaultRecordResult(checker Checker, result *types.Result, err error) {
```

Comment:
You can just use the name `RecordResult`. Later you need another function `RecordResultWithLabels()`

```suggestion
func RecordResult(checker Checker, result *types.Result, err error) {
```

Reply:
Renamed